pkg/services/ngalert/notifier/channels/sns.go

```diff
@@ -0,0 +1,218 @@
+package channels
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/grafana/grafana-aws-sdk/pkg/awsds"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+)
+
+// SNSNotifier is responsible for sending
+// alert notifications to AWS SNS
+type SNSNotifier struct {
+	*Base
+	tmpl                  *template.Template
+	SnsTopic              string
+	Subject               string
+	Body                  string
+	Message               string
+	MessageFormat         string
+	RemoveAlertData       bool
+	SNSClient             *sns.SNS
+	AWSDatasourceSettings *awsds.AWSDatasourceSettings
+	log                   log.Logger
+}
+
+// NewSNSNotifier is the constructor for the SNS notifier
+func NewSNSNotifier(model *NotificationChannelConfig, t *template.Template, fn GetDecryptedValueFn) (*SNSNotifier, error) {
+	snsLogger := log.New("unified.notifier.sns")
+	topicArn := model.Settings.Get("topic").MustString()
+
+	// parse the region out of the ARN example: arn:aws:sns:us-west-2:479104307918:GrafanaTestUSWEST2
+	arnSlice := strings.Split(topicArn, ":")
+	if len(arnSlice) != 6 {
+		return nil, receiverInitError{Reason: "Invalid topic ARN provided."}
+	}
+
+	var externalID, accessKey, secretKey, credentials string
+	region := arnSlice[3]
+	authType := model.Settings.Get("authProvider").MustString()
+	subject := model.Settings.Get("subject").MustString("AWS Grafana - Alert")
+	body := model.Settings.Get("body").MustString()
+	messageFormat := model.Settings.Get("messageFormat").MustString("text")
+	message := model.Settings.Get("text").MustString(`{{ template "default.message" . }}`)
+
+	at := awsds.AuthTypeDefault
+	switch authType {
+	case awsds.AuthTypeSharedCreds.String():
```

Comment:
I think it would be better to make `authType` an `awsd.AuthType` with `func ToAuthType(authType string) (AuthType, error)` instead of using `.String()` for each case.